L04-generics/src/main/java/homework/CustomerService.java

```diff
@@ -1,21 +1,21 @@
 package homework;
 
 import java.util.Map;
+import java.util.TreeMap;
 
-@SuppressWarnings({"java:S1186", "java:S1135", "java:S1172"}) // при выполнении ДЗ эту аннотацию надо удалить
 public class CustomerService {
-
-    // todo: 3. надо реализовать методы этого класса
-    // важно подобрать подходящую Map-у, посмотрите на редко используемые методы, они тут полезны
+    private TreeMap<Customer, String> customerData = new TreeMap<>();
```

Comment:
Тип лучше объявлять минимальным интерфейсом.

```suggestion
    private NavigableMap<Customer, String> customerData = new TreeMap<>();
```